math/src/elliptic_curve/short_weierstrass/curves/bls12_381/compression.rs

```diff
@@ -19,6 +19,11 @@ type G2Point = ShortWeierstrassProjectivePoint<BLS12381TwistCurve>;
 type BLS12381FieldElement = FieldElement<BLS12381PrimeField>;
 
 impl Compress for BLS12381Curve {
+    /// This functionality includes the compression and decompression for points belonging to the BLS12-381, following the ideas in
+    /// Zcash curve compression, check https://hackmd.io/@benjaminion/bls12-381#Point-compression and https://github.com/zcash/librustzcash/blob/6e0364cd42a2b3d2b958a54771ef51a8db79dd29/pairing/src/bls12_381/README.md#serialization
+    /// The way we encode points is different from the one used ordinarily for serialization in lambdaworks.
+    /// G1 points are represented by their x coordinate in big-endian form (48 bytes), with the 3 most significant bits used to give information on the compressed format, whether the point is the point at infinity and which of the two roots to take
+    /// G2 points are represented by their x coordinate in big-endian form (96 bytes), following the order a * i + b. The three most significant bits contain the same type of information as in G1.
```

Comment:
This should be on the Impl compress, else it's the description of the g1 point

```suggestion
    /// This functionality includes the compression and decompression for points belonging to the BLS12-381, following the ideas in
    /// Zcash curve compression, check https://hackmd.io/@benjaminion/bls12-381#Point-compression and https://github.com/zcash/librustzcash/blob/6e0364cd42a2b3d2b958a54771ef51a8db79dd29/pairing/src/bls12_381/README.md#serialization
    /// The way we encode points is different from the one used ordinarily for serialization in lambdaworks.
    /// G1 points are represented by their x coordinate in big-endian form (48 bytes), with the 3 most significant bits used to give information on the compressed format, whether the point is the point at infinity and which of the two roots to take
    /// G2 points are represented by their x coordinate in big-endian form (96 bytes), following the order a * i + b. The three most significant bits contain the same type of information as in G1.
```